packages/flutter/lib/src/painting/text_span.dart

```diff
@@ -345,7 +354,7 @@ class TextSpan extends DiagnosticableTree {
         && typedOther.style == style
         && typedOther.recognizer == recognizer
         && typedOther.semanticsLabel == semanticsLabel
-        && listEquals<TextSpan>(typedOther.children, children);
+        && listEquals<InlineSpan>(typedOther.children, children);
   }
 
   @override
```

Comment:
nit: while you're here, please change hashCode's order to match the order in operator==